src/pages/index.js

```diff
@@ -21,80 +21,271 @@ import {
   cardList,
   cardOpenModal,
 } from "../utils/constants.js";
+import Api from "../components/Api.js";
+import PopupWithConfirm from "../components/PopupWithConfirm.js";
 
-//CARD FUNCTIONS
+//-----------------------------------------------------------API---------------------------------------------------------------------------------------------------------------------
+//api instance
 
-function createCard(cardData) {
-  const card = new Card(cardData, "#card", handleCardClick).generateCard();
-  return card;
+const api = new Api({
+  baseUrl: "https://around.nomoreparties.co/v1/cohort-3-en",
+  headers: {
+    authorization: "6960473d-b35e-4d3c-8ef0-2e694acf306b",
+    "Content-Type": "application/json",
+  },
+});
+
+let userId;
+let cardListSection;
+
+Promise.all([api.getInitialCards(), api.getUserInfo()])
+  .then(([initialCards, userData]) => {
+    userId = userData._id;
+    console.log(userId);
+    userInfo.setUserInfo(userData.name, userData.about);
+    userInfo.setProfileAvatar(userData.avatar);
+    cardListSection = new Section(
+      {
+        items: initialCards,
+
+        renderer: ({ name, link, likes, _id, owner }) => {
+          const newCard = createCard({
+            name,
+            link,
+            likes,
+            _id,
+            userId,
+            ownerId: owner._id,
+          });
+          cardListSection.addItem(newCard);
+        },
+      },
+      cardList
+    );
+    cardListSection.renderItems();
+  })
+  .catch(() => (err) => console.log(err));
+//--------------------------------------------------------------AVATAR-------------------------------------------------------------------------------------
+//new avatar constants
+
+const profileAvatarModalImageSelector = document.querySelector(
+  // "#update-avatar-modal"
+  ".profile__image"
+);
+const profileAvatarModalSelector = document.querySelector(
+  "#update-avatar-modal"
+);
+const profileAvatarPeniclIcon = document.querySelector(
+  ".profile__icon-edit-button"
+);
+
+//avatar form validator instance
+
+const avatarFormValidator = new FormValidator(
```

Comment:
**Could be improved**

You can **universally** create instances of validators for all forms in the project storing them inside one object `formValidators`.  And then you can take any validator using attribute `name` of the form where you need to disable/enable the submit button or remove errors.

```js
const formValidators = {}

// enable validation
const enableValidation = (config) => {
  const formList = Array.from(document.querySelectorAll(config.formSelector))
  formList.forEach((formElement) => {
    const validator = new FormValidator(config, formElement)
    // here you get the name of the form
    const formName = formElement.getAttribute('name')

   // here you store a validator by the `name` of the form
    formValidators[formName] = validator;
   validator.enableValidation();
  });
};

enableValidation(config);

```

And now you can use them for disabling buttons or clearing errors:

```js

formValidators[ profileForm.getAttribute('name') ].resetValidation()

// or you can use a string – the name of the form (you know it from `index.html`)

formValidators['profile-form'].resetValidation()

```